doc/kernel/object_cores/index.rst

```diff
@@ -0,0 +1,234 @@
+.. _object_cores_api:
+
+Object Cores
+############
+
+Object cores are a kernel debugging tool that can be used to both identify and
+perform operations on registered objects.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Object Core Concepts
+********************
+
+Each instance of an object has embedded within it an object core field named
+`obj_core`. Objects of the same type are linked together via their respective
+object cores to form a singly linked list. Each object core also links to the
+their respective object type. Each object type contains a singly linked list
+linking together all the object cores of that type. Object types are also
+linked together via a singly linked list. Together, this can allow debugging
+tools to traverse all the objects in the system.
+
+Object cores have been integrated into following kernel objects:
+ * Condition Variables
+ * Events
+ * FIFOs and LIFOs
+ * Mailboxes
+ * Memory Slabs
+ * Message Queues
+ * Mutexes
+ * Pipes
+ * Semaphores
+ * Timers
+ * System Memory Blocks
```

Comment:
the leading white spaces is causing the list to render as a definition list, so you need to drop it. Also would be nice to link to doc pages (I've added explicit aliases for the pages to keep the list easy to read when browsing the source.

```suggestion
Object cores have been integrated into following kernel objects:

* :ref:`Condition Variables <condvar>`
* :ref:`Events <events>`
* :ref:`FIFOs <fifos_v2>` and ref:`LIFOs <lifos_v2>`
* :ref:`Mailboxes <mailboxes_v2>`
* :ref:`Memory Slabs <memory_slabs_v2>`
* :ref:`Message Queues <message_queues_v2>`
* :ref:`Mutexes <mutexes_v2>`
* :ref:`Pipes <pipes_v2>`
* :ref:`Semaphores <semaphores_v2>`
* :ref:`Timers <timers_v2>`
* :ref:`System Memory Blocks <sys_mem_blocks>`
```

Reply:
Fixing.

Reply:
> Fixing.

thx! If you have a chance please test locally: I might have gotten some references wrong as i provided my feedback through GitHub UI and did not actually test this change